include/deal.II/base/polynomials_bdm.h

```diff
@@ -203,14 +196,6 @@ class PolynomialsBDM : public TensorPolynomialsBase<dim>
 };
 
 
-template <int dim>
-inline unsigned int
-PolynomialsBDM<dim>::degree() const
-{
-  return polynomial_space.degree() - 1;
```

Comment:
And the removal of this function is essentially just a clean-up to make sure the class doesn't have a function that was awkwardly named? (Because it confuses the "polynomial degree" with the "degree of the element".)

This makes sense to me. It's unfortunate that we say things like "the RT element of degree 0" (or the "RT(0) element") as that has polynomial degree 1.

Reply:
Yes and I didn't clean it up initially because I was uncertain what this `PolynomialsBDM<dim>::degree()` function did. `polynomial_space.degree() - 1` did not make sense until I took a few minutes to read how `polynomial_space` was constructed and how `degree()` was defined for that a `polynomial_space` object.

It is a little unfortunate. Even the BR1 element is a polynomial degree 2 space, but I believe this is okay because the finite element will always keep track of the space degree `k`, even if the polynomial space has a different degree like `k+1`.